app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetController.scala

```diff
@@ -57,7 +58,15 @@ class EditSetController @Inject() (
           value => value.length <= 8000
         )
         .verifying(messagesApi("edit-set.record.missing.scope-and-content")(Lang.apply("en")), _.nonEmpty),
-      "coveringDates" -> text,
+      "coveringDates" -> text
+        .verifying(
+          messagesApi("edit-set.record.missing.covering-dates")(Lang.apply("en")),
+          _.trim.nonEmpty
+        )
+        .verifying(
+          messagesApi("edit-set.record.error.covering-dates")(Lang.apply("en")),
+          value => CoveringDateCalculator.getStartAndEndDates(value).isRight
+        ),
```

Comment:
The spec states that the field has a 255 character limit. I think we should be checking this here somewhere as currently I am able to enter more than 255 characters which are otherwise valid, e.g. by having a sequence of date ranges like 1801-1802;1804-1805;1807-1808.. etc